### ISSUE TITLE
Screen reader does not announce the expand state of select box

### ISSUE THREAD

**Issue by RohitJadhav-TCS** [microsoft/vscode#76869]
**Environment Details:**
VSCode Version : 1.36.0 (user setup)
OS Version : Win10

GitHubTags: #A11yMAS; #A11yTCS; #Win32; #DesktopApp; #A11y_VSCodeEditor; #MAS1.3.1 #A11ySev2; #Benchmark 

**Additional Details:**
MAS Violated : MAS 1.3.1
Screen Reader Versions : NVDA2019.1.1 JAWS2019

**Repro Steps:**
1.Open Visual studio code editor.
2.Using "Ctrl+`" key open the Integrated Terminals.
3.Navigate to terminal action toolbar.
4.Navigate to combo box and expand it using alt+down arrow key.

**Actual:**
Screen reader does not announces the expand state of terminal selection combo box.

This issue has been observed in "Win 10 + JAWS" and "Win10 + NVDA".

**Expected:**
Screen reader should announce the expand and collapse state of combo box.

**Recommendations:**
Refer below link which is repository of bug fixes code snippets:
[https://microsoft.sharepoint.com/teams/msenable/mas/pages/browse-fixes.aspx](https://microsoft.sharepoint.com/teams/msenable/mas/pages/browse-fixes.aspx)

**User Impact:**
If the expand state of combo box are not announced then the screen reader users will not get the exact information.

**MAS Reference:**
MAS 1.3.1 - [https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={54f28d1f-a2d1-4dcd-84e1-5c9b87e8aba4}](https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={54f28d1f-a2d1-4dcd-84e1-5c9b87e8aba4})

**Attachment for Reference:**
[VSCode_Integrated Terminal_ScreenReader does not announce the expand state of combo box.pptx](https://github.com/microsoft/vscode/files/3368048/VSCode_Integrated.Terminal_ScreenReader.does.not.announce.the.expand.state.of.combo.box.pptx)

**Comment by isidorn** [comment]
I acknowledge that the 'expanded' state is not properly read out by the screen reader.
We only set the collapsed state, however not the expanded state.
We are acceptin PRs that fix this.
fyi @cleidigh 

Code pointer https://github.com/Microsoft/vscode/blob/2467eab1338d03631cb4e11123b4ded84615aa0e/src/vs/base/browser/ui/selectBox/selectBoxCustom.ts#L83

**Comment by RohitJadhav-TCS** [comment]
Similar issue is observed in below scenario;
1.Validate the Output panel can show log output
Navigation step: I Open Visual studio code editor  
                           II Press ctrl+shift+u
                          III Validate the Output panel can show log output

**Comment by isidorn** [comment]
@RohitJadhav-TCS thanks for commenting here. Yes they both use the same underlying component, thus if we fix this in the underlying select box compoentn it will be fixed everywhere. Due to that having one issue is best.

**Comment by cleidigh** [comment]
@isidorn 
I could pick this up, we have had similar issues before.

**Comment by cleidigh** [comment]
@isidorn 
I can reproduce the issue with NVDA.
Currently I cannot build the latest code, I'm hoping this is a temporary glitch.
I can build some of my old branches attempting to see if and where there was a regression.
As I said before, this is been a back-and-forth issue.  If you look at the issue below
the expanded state was okay ,however, you will notice this was with a tree view.

I tried the trick of adding the role again, but this did not work.
If you use the command dropdown , you'll hear the output for levels , but no expanded
indication.  I know there have been many changes to tree/list given the prior statement
do you think the issue may be there?



https://github.com/microsoft/vscode/issues/61700

**Comment by cleidigh** [comment]
@isidorn 
do you think this has to do with the list/tree component?

**Comment by isidorn** [comment]
@cleidigh sorry for not responding sooner I was on vacation.
Yes I believe that the "expanded" state gets properly read out when using a native select box. However when using a list/tree undearneath it is not read out.
I belive the fix would be to properly set the state `aria-expanded` to the custom select box.
I see we are alrady doing something like that [here](https://github.com/Microsoft/vscode/blob/980ad1ce090d58fdac487ad376612ae6e2f23134/src/vs/base/browser/ui/selectBox/selectBoxCustom.ts#L759) however we might need to do this on another HTML element. It would take some experimenting with the screen reader.

**Comment by isidorn** [comment]
To verify:
1. Use windows, keyboard navigate to a select drop down, like the output channels dropdown or the terminal dropdown
2. Make sure that when you expand it the `aria-expanded` is properly set on the HTML element

**Comment by HindujaArugula** [comment]
As per the confirmation that the issue will not be there in 2023 so closing the bug.
Github Tags:#Closed;
